cmd/goal/clerk.go

```diff
@@ -1358,3 +1362,17 @@ func decodeTxnsFromFile(file string) []transactions.SignedTxn {
 	}
 	return txgroup
 }
+
+func traceCmdOptionToSimulateTraceConfigModel() *model.SimulateTraceConfig {
+	simulateTraceConfig := model.SimulateTraceConfig{}
+	constTrue := true
+
+	if simulateExecTraceOptIn {
+		simulateTraceConfig.Enable = &constTrue
+	}
+
+	if simulateTraceConfig == (model.SimulateTraceConfig{}) {
+		return nil
+	}
+	return &simulateTraceConfig
+}
```

Comment:
It seems as though this code is going out of its way to keep the request empty unless the trace is requested. I know we often do stuff like this in the protocol, it's sort of like `omitempty`. But I think sending an extra `false`, if indeed that would happen is fine, in exchange for clearer code.
```suggestion
func traceCmdOptionToSimulateTraceConfigModel() *model.SimulateTraceConfig {
	return model.SimulateTraceConfig{
	   Enable: &simulateExecTraceOptIn
	}
}
```
It seems like it's ok to always return the object, and always fill it.